workflow/lib/preprocess/preprocess.py

```diff
@@ -90,22 +96,22 @@ def nd2_to_tiff(file: str, channel_order_flip: bool = False, verbose: bool = Fal
     """
     # Load with xarray to keep dimension labels
     image = nd2.imread(file, xarray=True)
-    
+
     if verbose:
         print(f"Original dimensions: {image.dims}")
-       
+
     # Handle Z-stack if present
-    if 'Z' in image.dims:
-        image = image.max(dim='Z')
-    
+    if "Z" in image.dims:
+        image = image.max(dim="Z")
+
     # Convert to numpy array in CYX order
-    image_array = image.transpose('C', 'Y', 'X').values
+    image_array = image.transpose("C", "Y", "X").values
 
     # Flip channel order on C axis if requested
     if channel_order_flip:
         image_array = np.flip(image_array, axis=0)  # axis 0 is the C axis
 
     if verbose:
         print(f"Final dimensions: {image_array.shape}")
-    
```

Comment:
There aren't any noticeable changes here. I think I just need to retest it. I don't want to revert back to nd2Reader. you don't need this data anyway for the time being...

One thing to think about--the preprocess takes each nd2 file and creates a tiff. However, it takes a group of nd2 files to create the metadata. This is a bit annoying, and also requires us to have different functions for processing metadata in a tile or well based format. It may make more sense to have one metadata file per each nd2 file, and then have a combine_metadata step. This may also assist with debugging, given your error message. Let's discuss this later.

Reply:
> you don't need this data anyway for the time being...

Yep! 

> It may make more sense to have one metadata file per each nd2 file, and then have a combine_metadata step. This may also assist with debugging, given your error message. Let's discuss this later.

This definitely sounds like a good idea.